### ISSUE TITLE
Resolve relationship between context and capabilities

### ISSUE THREAD

**Issue by danielbachhuber** [WP-API/WP-API#702]
While working on #701, I realized that `context = edit` can be passed to any `GET /wp/users` request to get full user details. Which, in hindsight, makes sense, because if a user has the `list_users` cap, they conceivably can get all of those details out of the Manage Users table.

However, we'll shortly need to remove the current `list_users` cap check at the beginning of each method because a basic amount of user data will need to be exposed if the user has at least one published post (#695).

The `master` branch has `json_check_post_permission()`, which no one is really a fan of.

In short, the relationship between contexts and capabilities is a bit muddy right now. We should make it much clearer.

Maybe we should reintroduce `check_context_permission()` for controllers? https://github.com/WP-API/WP-API/pull/421/files#diff-08b662d17a90b99fbd5dc8592dafd6c9R124

Related #605

**Comment by danielbachhuber** [comment]
With #1049, I'm much happier with things now. Contexts are defined to a field in the schema, which makes it much easier to see when they're exposed. Also, with the abstraction we've been working on, contexts are much more closely correlated with capabilities.

**Comment by danielbachhuber** [comment]
@joehoyle Any thoughts on remaining steps for this?

**Comment by joehoyle** [comment]
I don't really have anything to add here - I like the abstraction with putting context in the schema, but beyond that I'm not convinced if we can "resolve the relationship" for much more of a degree. The more I work the capabilities, the more I come to the conclusion that they can't be done declaratively, despite the "declarative" api they are meant to do. Having declarative caps for different contexts isn't realistic I think, so I'm happy with the procedural checks we have, which is now a lot saner since your most recent PR on this ticket.

**Comment by danielbachhuber** [comment]
I'm pretty happy with the abstractions we have now (permission callback and context as a part of a schema). I think we start the conversation again in the future as merited.
